libcalico-go/lib/backend/k8s/conversion/conversion.go

```diff
@@ -262,12 +263,20 @@ func (c converter) K8sNetworkPolicyToCalico(np *networkingv1.NetworkPolicy) (*mo
 	// This order might change in future.
 	order := float64(1000.0)
 
+	var convErr cerrors.ErrorPolicyConversion
+	var convErrRules []cerrors.ErrorPolicyConversionRule
+
 	// Generate the ingress rules list.
 	var ingressRules []apiv3.Rule
 	for _, r := range np.Spec.Ingress {
 		rules, err := c.k8sRuleToCalico(r.From, r.Ports, np.Namespace, true)
 		if err != nil {
 			log.WithError(err).Warn("dropping k8s rule that couldn't be converted.")
+			// Add rule to conversion error slice
+			convErrRules = append(convErrRules, cerrors.ErrorPolicyConversionRule{
+				IngressRule: &r,
+				Reason:      "k8s rule couldn't be converted",
```

Comment:
We have an `err` in-hand here, which means we should be able to give a more detailed reason than this by just including the original error.

Reply:
good point, thanks!